pyiode/iode_database/lists_database.pyx

```diff
@@ -104,5 +104,118 @@ cdef class Lists(_AbstractDatabase):
         else:
             self.database_ptr.add(key.encode(), value.encode())
 
+    def from_series(self, s: Series):
+        r"""
+        Copy the pandas Series `s` into the IODE Lists database.
+
+        Notes
+        -----
+        The index of the passed Series is sorted in alphabetical order before 
+        copying to IODE Lists database.
+
+        Examples
+        --------
+        >>> from iode import lists, split_list
+        >>> import pandas as pd
+        >>> lists.clear()
+        >>> len(lists)
+        0
+
+        >>> # create the pandas Series
+        >>> names = ["SEMICOLON_LST", "COMMA_LST", "WHITESPACE_LST", "TAB_LST", "MIX_LST"]
+        >>> semicolon_lst = "A;B;C;D;E;F"
+        >>> comma_lst = "A,B,C,D,E,F"
+        >>> whitespace_lst = "A B C D E F" 
+        >>> tab_lst = "A    B    C    D    E    F"
+        >>> mix_lst = "  A;B,C D    E;F  "
+        >>> data = [semicolon_lst, comma_lst, whitespace_lst, tab_lst, mix_lst]
+        >>> s = pd.Series(data=data, index=names, dtype=str, name="Lists")
+        >>> # display the pandas series
+        >>> s          # doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
+        SEMICOLON_LST                        A;B;C;D;E;F
+        COMMA_LST                            A,B,C,D,E,F
+        WHITESPACE_LST                       A B C D E F
+        TAB_LST               A    B    C    D    E    F
+        MIX_LST                         A;B,C D    E;F
+        Name: Lists, dtype: object
+
+        >>> # load into the IODE Lists database
+        >>> lists.from_series(s)
+        >>> len(lists)
+        5
+        
+        >>> lists.get_names()
+        ['COMMA_LST', 'MIX_LST', 'SEMICOLON_LST', 'TAB_LST', 'WHITESPACE_LST']
+        >>> lists["SEMICOLON_LST"]
+        'A;B;C;D;E;F'
+        >>> split_list(lists["SEMICOLON_LST"])
+        ['A', 'B', 'C', 'D', 'E', 'F']
+        >>> # note: leading and trailing whitespaces are automatically stripped 
+        >>> #       when copied into the Lists database
+        >>> mix_lst
+        '  A;B,C D    E;F  '
+        >>> lists["MIX_LST"]
+        'A;B,C D    E;F'
```

Comment:
> Is it intended that lists preserve their definition inside iode?

I don't think it is. Internally, when an IODE list has to be split into its elements, the C API use the function `SCR_vtoms()` which split a char* array into a char** table according to a list of predefined separators characters (`,;\s`). 

> I would probably lean in favor of normalization

That means I need to first split the IODE list as a Python list, then use `join(';')` to recreate a *formated* list. 
This will cost a little performance but adding or updating an IODE list is not something that users often do. 
Plus, with the Python built-in lists, the IODE lists will become somewhat obsolete.

> but it actually depends if users use/give some different meanings to the various ways to define lists (maybe as some kind of hierarchy???).

I don't think any current user mixes different separator characters in the same IODE list to define a kind of hierarchy but I cannot be sure.

One way or another, it doesn't hurt to normalize and it is reversible. So I would go for the normalization.
Thanks for the advice.